### ISSUE TITLE
[ILM] Delete phase doen't delete indexes.

### ISSUE THREAD

**Issue by zakabluk** [elastic/elasticsearch#68794]
ELK version 7.9.3.
My ILM policy which was added to index template instead of old policy. And new indexes created with them.
```
PUT _ilm/policy/delete-after-7-days
{
  "policy": {
    "phases": {
      "hot": {
        "min_age": "0ms",
        "actions": {
          "set_priority": {
            "priority": 100
          }
        }
      },
      "delete": {
        "min_age": "7d",
        "actions": {
          "delete": {
            "delete_searchable_snapshot": true
          }
```
New indexes has result
GET .ds-logstash-f-lb-2021.01.29-000001/_ilm/explain
```
  "indices" : {
    ".ds-logstash-fornex-lb-2021.01.29-000001" : {
      "index" : ".ds-logstash-f-lb-2021.01.29-000001",
      "managed" : true,
      "policy" : "delete-after-7-days",
      "lifecycle_date_millis" : 1611878426807,
      "age" : "11.38d",
      "phase" : "delete",
      "phase_time_millis" : 1612483233516,
      "action" : "delete",
      "action_time_millis" : 1612483233516,
      "step" : "delete",
      "step_time_millis" : 1612483781849,
      "phase_execution" : {
        "policy" : "delete-after-7-days",
        "phase_definition" : {
          "min_age" : "7d",
          "actions" : {
            "delete" : {
              "delete_searchable_snapshot" : true
            }
          }
        },
        "version" : 4,
        "modified_date_in_millis" : 1612254154538
      }
```
but indexes dosen't deleted. In logs I see error
```
"stacktrace": ["java.lang.IllegalStateException: index [.ds-logstash-f-lb-2021.01.29-000001] is the write index for data stream [logstash-f-lb-2021.01.29]. stopping execution of lifecycle [delete-after-7-days] as a data stream's write index cannot be deleted. manually rolling over the index will resume the execution of the policy as the index will not be the data stream's write index anymore",
"at 
```

**Comment by dakrone** [comment]
Thanks for opening this @zakabluk, while usually someone would use a `rollover` action in the policy, it's still valid not to have one.

In the delete step we should detect if the index is a singular index in a data stream, and delete the data stream (instead of the index) if so.

**Comment by dakrone** [comment]
I fixed this in #69637, I just forgot to link this issue, so I'm closing it now.
